lib/include/nitro/gx/g3.h

```diff
@@ -147,20 +150,77 @@ typedef enum {
     GX_MTXMODE_TEXTURE = 3
 } GXMtxMode;
 
-#define GX_PACK_VIEWPORT_PARAM(x1, y1, x2, y2)  ((u32)((x1) | ((y1) << 8) | ((x2) << 16) | ((y2) << 24)))
+typedef enum {
+	GX_TEXGEN_NONE        = 0,
+	GX_TEXGEN_TEXCOORD    = 1,
+	GX_TEXGEN_NORMAL      = 2,
+	GX_TEXGEN_VERTEX      = 3
+} GXTexGen;
+
+typedef enum {
+	GX_TEXREPEAT_NONE    = 0,
+	GX_TEXREPEAT_S       = 1,
+	GX_TEXREPEAT_T       = 2,
+	GX_TEXREPEAT_ST      = 3
+} GXTexRepeat;
+
+typedef enum {
+	GX_TEXFLIP_NONE    = 0,
+	GX_TEXFLIP_S       = 1,
+	GX_TEXFLIP_T       = 2,
+	GX_TEXFLIP_ST      = 3
+} GXTexFlip;
 
 typedef enum {
-    GX_POLYGONMODE_MODULATE = 0,
-    GX_POLYGONMODE_DECAL = 1,
-    GX_POLYGONMODE_TOON = 2,
-    GX_POLYGONMODE_SHADOW = 3
+	GX_BEGIN_TRIANGLES      = 0,
+	GX_BEGIN_QUADS          = 1,
+	GX_BEGIN_TRIANGLE_STRIP = 2,
+	GX_BEGIN_QUAD_STRIP     = 3
+} GXBegin;
+
+typedef enum {
+	GX_POLYGONMODE_MODULATE    = 0,
+	GX_POLYGONMODE_DECAL       = 1,
+	GX_POLYGONMODE_TOON        = 2,
+	GX_POLYGONMODE_SHADOW      = 3
 } GXPolygonMode;
 
 typedef enum {
-    GX_CULL_ALL = 0,
-    GX_CULL_FRONT = 1,
-    GX_CULL_BACK = 2,
-    GX_CULL_NONE = 3
+	GX_CULL_ALL      = 0,
+	GX_CULL_FRONT    = 1,
+	GX_CULL_BACK     = 2,
+	GX_CULL_NONE     = 3
 } GXCull;
 
+#ifdef SDK_ADS
+    typedef struct {
```

Comment:
what thr fuck is an SDK_ADS????
this entire struct def doesn't follow the style either
indentation, struct name, also don't embed structs like this